docs/upload_specifications.ADR.md

```diff
@@ -0,0 +1,170 @@
+delivering the templates and instructions to users will be developed.
+
+Currently, for all in scope apps, the inputs are all strings, numbers, or booleans. There are
+no unusual inputs such as grouped parameters or dynamic lookups. Including future upload apps
+with these features in CSV-based upload may require additional engineering.
+
+## User operations
+
+* The user uploads the upload specification files to the staging area along with all the files
+  inluded in the specification. 
+* The user selects the `Upload Specification` type for the specification files.
+  * The user may also select other files in the staging area to include in the upload along
+    with the files listed in the specification.
+    * The user *does not* have to select any files included in the specification.
+* The user clicks `Import Selected`.
+
+As such, a new type must be added to the StS: `upload_specification` with the title
+`Upload Specification`. *Nota bene*: It may be preferable to have the Narrative specify the
+titles to display in the staging area dropdown rather than the StS.
+
+## Narrative Bulk Upload cell operations
+
+When the bulk upload cell sees a `upload_specification` data type, it calls out to the StS
+endpoint detailed below to get the parsed specifications, and then autofills itself with
+those specifications.
+
+## Staging service upload specification endpoint
+
+The StS endpoint responds to an HTTP GET with a file list in a `files` URL parameter. It is
+extremely unlikely that there will be a large enough set of data types that URL length limits
+will be reached so this seems adequate. The StS will respond with the contents of the files
+parsed into a structure that is similar to that of the input spec, mapped by data type:
+
+```
+Headers:
+Authorization: <KBase token>
+
+GET /bulk_specification/?files=<file1.csv[,file2.csv,...]>
+
+Response:
+{"types": {
+    "gff_metagenome": [
+        {"fasta_file": "mygenome.fa",
+         "gff_file": "mygenome.gff3",
+         "genome_name": "mygenomeobject",
+         "source": null,
+         "release": null,
+         "genetic_code: 11,
+         "generate_missing_genes: 0
+         },
+        {"fasta_file": "mygenome2.fa",
+         "gff_file": "mygenome2.gff3",
+         "genome_name": "mygenomeobject2",
+         "source": "yermumspoo",
+         "release": 30456,
+         "genetic_code: 11,
+         "generate_missing_genes: 1
+         },
+         ...
+    ],
+    "sra_reads": [
+        ...
+    ]
+    ...
+}
+```
+
+The order of the input structures MUST be the same as the order in the input files.
+
+Notably, the service will provide the contents of the files as is and will not perform most error
+checking, including for missing or unknown input parameters. Most error checking will be performed
+in the bulk upload cell configuration tab like other uploads, allowing for a consistent user
+experience.
+
+### Missing files
```

Comment:
Do we need an error code for permissions issues as well?

Reply:
I don't think it's possible to ask for files that aren't yours is it? The service will only look in the directory for the user, right?

Reply:
No, I mean if the file  or the file it references is unreadable due to a permissions issue.

Reply:
Oh I see, like the old workspace_export problems. I added a catch all error structure for unexpected errors like that